Tests/MessagingInApp/Views/InAppMessageViewTest.swift

```diff
@@ -221,6 +221,32 @@ class InAppMessageViewTest: UnitTest {
         XCTAssertNil(getInAppMessage(forView: inlineView))
     }
 
+    @MainActor
+    func test_expiration_givenExpiredMessageNotYetDisplayed_expectDoNotDisplayMessage() async {
+        let givenMessageDisplayed = Message(elementId: .random)
+        let givenMessageThatExpires = Message(elementId: .random)
+        queueMock.getInlineMessagesReturnValue = [givenMessageDisplayed, givenMessageThatExpires]
+
+        let inlineView = InAppMessageView(elementId: givenMessageDisplayed.elementId!)
+
+        await onDoneRenderingInAppMessage(givenMessageDisplayed, insideOfInlineView: inlineView)
+
+        XCTAssertTrue(isInlineViewVisible(inlineView))
+        XCTAssertEqual(getInAppMessage(forView: inlineView), givenMessageDisplayed)
+
+        // Simulate message expiration.
+        await simulateSdkFetchedMessages([givenMessageDisplayed])
+
+        // Expect still showing the same message as before the fetch call.
+        XCTAssertEqual(getInAppMessage(forView: inlineView), givenMessageDisplayed)
+
+        await onCloseActionButtonPressed(onInlineView: inlineView)
+
+        // Expect we do not show the expired message but instead close the View.
+        XCTAssertFalse(isInlineViewVisible(inlineView))
+        XCTAssertNil(getInAppMessage(forView: inlineView))
```

Comment:
If expired message is not displayed then is it removed from the queue ? This test focuses on what is displayed to the user but wondering if what isn't displayed stays in the queue or is removed and is covered in another test case?

Reply:
> If expired message is not displayed then is it removed from the queue ?

The local SDK queue will not contain a message if it is expired. So, yes, if a message is not displayed, it should also not be in the local queue. 

The backend server is what expires messages. When the mobile SDK performs a fetch for messages from the backend, if the backend does not return a message, it could be because the message has expired. 

the function 

```swift
await simulateSdkFetchedMessages([givenMessageDisplayed])
```

in the test function is what modifies the SDK's queue. Because `givenMessageThatExpires` is not included in the `simulateSdkFetchedMessages` call, then that message will not be in the SDK's local queue.